Nos/Models/NosNotification+CoreDataClass.swift

```diff
@@ -16,16 +16,21 @@ public class NosNotification: NSManagedObject {
 
     class func createIfNecessary(
         from eventID: HexadecimalString, 
-        authorKey: HexadecimalString, 
+        date: Date?,
+        authorKey: HexadecimalString,
         in context: NSManagedObjectContext
     ) throws -> NosNotification? {
+        guard let date, date > cutoffDate() else {
```

Comment:
NotificationsView queries the database for events and build "notification cards" for these events. It doesn't use NosNotification objects at all. These are used for alerting the user with a badge using the notification center.

Reply:
So what that guard does is to prevent alerting the user for an event that happened more than two months ago. because as we now clean old notification objects we don't know if the user was already alerted or not.